File: Examples/VisualBasic/Pages/RollBackRevisions.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic

Namespace Pages
    Public Class RollBackRevisions
        Public Shared Sub Run()
            ' ExStart:RollBackRevisions
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Pages()

            ' Load OneNote document and get first child           
            Dim document As New Document(dataDir & Convert.ToString("Aspose.one"))
            Dim page As Page = document.FirstChild
            Dim lastPage As Page = Nothing
            For Each pageRevision As Page In document.GetPageHistory(page)
                lastPage = pageRevision
            Next
            document.RemoveChild(page)

            document.AppendChild(lastPage)

            document.Save(dataDir & "RollBackRevisions_out_.one")
            ' ExEnd:RollBackRevisions
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Pages/RollBackRevisions.cs
using System.IO;
using Aspose.Note;
using System;
using System.Collections.Generic;

namespace Aspose.Note.Examples.CSharp.Pages
{
    using System.Linq;

    public class RollBackRevisions
    {
        public static void Run()
        {
            // ExStart:RollBackRevisions
            // ExFor:Document
            // ExFor:Document.GetPageHistory
            // ExSummary:Shows how to restore previous version of a page.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Pages();

            // Load OneNote document and get first child           
            Document document = new Document(dataDir + "Aspose.one");
            Page page = document.FirstChild;           
            Page previousPageVersion = document.GetPageHistory(page).Last();

            document.RemoveChild(page);
            document.AppendChildLast(previousPageVersion);

            document.Save(dataDir + "RollBackRevisions_out.one");

            // ExEnd:RollBackRevisions
        }
    }
}